web/src/main/java/com/epam/esm/web/controller/TagController.java

```diff
@@ -28,38 +37,48 @@
 
     private final TagServiceImpl tagServiceImpl;
     private final CertificateServiceImpl certificateServiceImpl;
+    private final TagDTOMapper tagDTOMapper;
+    private final CertificateDTOMapper certificateDTOMapper;
 
     @Autowired
     public TagController(TagServiceImpl tagServiceImpl,
-                         CertificateServiceImpl certificateServiceImpl) {
+                         CertificateServiceImpl certificateServiceImpl,
+                         CertificateDTOMapper certificateDTOMapper,
+                         TagDTOMapper tagDTOMapper) {
         this.tagServiceImpl = tagServiceImpl;
         this.certificateServiceImpl = certificateServiceImpl;
+        this.certificateDTOMapper = certificateDTOMapper;
+        this.tagDTOMapper = tagDTOMapper;
     }
 
     @GetMapping
-    public ResponseEntity<?> readAll(@RequestParam Map<String, String> params)
-            throws ServiceException, BadRequestException {
-        Collection<Tag> list;
+    public ResponseEntity<?> read(@RequestParam MultiValueMap<String, String> params)
+            throws BadRequestException, NotFoundException {
+        List<TagDTO> list;
```

Comment:
it's better to create object: List<TagDTO> list = new ArrayList<>();

Reply:
Should we? create object only to fill the reference which soon will be aasigned to a new object?
List<Tag> tags;
        if (CollectionUtils.isEmpty(params)) {
            tags = tagServiceImpl.readAll();
        } else {
            tags = tagServiceImpl.read(params);
        }

Reply:
List list = new ArrayList<>();